src/multisignature.cpp

```diff
@@ -0,0 +1,248 @@
+// Copyright (c) 2017-2018 The PIVX developers
+// Distributed under the MIT/X11 software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <regex>
+#include <boost/algorithm/string/split.hpp>
+#include "multisignature.h"
+#include "utilstrencodings.h"
+#include "tinyformat.h"
+#include "util.h"
+#include "streams.h"
+#include "wallet.h"
+
+
+using namespace std;
+
+CMultisignatureAddress::CMultisignatureAddress(int nSignaturesRequired, const vector<string>& vAddressOwners)
+{
+    nSigsRequired = nSignaturesRequired;
+    nOwners = (int)vAddressOwners.size();
+
+    if(!ValidateConfiguration()) {
+        return;
+    }
+
+    if(!ConvertAndValidatePubKeys(vAddressOwners)){
+        return;
+    }
+
+    CreateRedeemScript();
+}
+
+
+CMultisignatureAddress::CMultisignatureAddress(string strRedeemScript)
+{
+    ParseRedeemScript(strRedeemScript);
+}
+
+bool CMultisignatureAddress::ValidateConfiguration()
+{//gather pub keys
```

Comment:
misplaced comment.

Either have it on it's own line inside the function, or use Doxygen compatible comments preceding the function